README.md

```diff
@@ -1,43 +1,470 @@
+    "collaborators" => 0,
+    "name" => "pro",
+    "private_repos" => 9999,
+    "space" => 976562499
+  },
+  node_id: "MDQ6VXNlcjE5NDQwMA==",
+  created_at: "2010-02-02T08:44:49Z",
+  blog: "http://www.dwyl.io/",
+  type: "User",
+  bio: "Learn something new every day. github.com/dwyl/?q=learn",
+  following_url: "https://api.github.com/users/nelsonic/following{/other_user}",
+  repos_url: "https://api.github.com/users/nelsonic/repos",
+  total_private_repos: 5,
+  html_url: "https://github.com/nelsonic",
+  public_gists: 29,
+  avatar_url: "https://avatars3.githubusercontent.com/u/194400?v=4",
+  organizations_url: "https://api.github.com/users/nelsonic/orgs",
+  url: "https://api.github.com/users/nelsonic",
+  followers: 2778,
+  updated_at: "2020-02-01T21:14:20Z",
+  location: "London",
+  hireable: nil,
+  name: "Nelson",
+  owned_private_repos: 5,
+  company: "@dwyl",
+  email: "nelson@gmail.com",
+  two_factor_authentication: true,
+  starred_url: "https://api.github.com/users/nelsonic/starred{/owner}{/repo}",
+  id: 194400,
+  following: 173,
+  login: "nelsonic",
+  collaborators: 8
+}
+```
+
+More info: https://developer.github.com/v3/users
+
+You can use this data however you see fit.
+(_obviously treat it with respect,
+  only store what you need and keep it secure_)
+
+
+## 4. Add the `/auth/github/callback` to `router.ex`
+
+Open your `lib/app_web/router.ex` file
+and locate the section that looks like `scope "/", AppWeb do`
+
+Add the following line:
+
+```elixir
+get "/auth/github/callback", GithubAuthController, :index
+```
+
+That will direct the API request response
+to the `GithubAuthController` `:index` function we defined above.
+
+> Example:
+[/lib/app_web/router.ex#L20](https://github.com/dwyl/elixir-auth-github-demo/blob/8dd8f7dd7cc43a9a7e918dd11c920833e17ef0af/lib/app_web/router.ex#L20)
+
+
+## 5. Update `PageController.index`
+
+In order to display the "Sign-in with GitHub" button in the UI,
+we need to _generate_ the URL for the button in the relevant controller,
+and pass it to the template.
+
+Open the `lib/app_web/controllers/page_controller.ex` file
+and update the `index` function:
+
+From:
+```elixir
+def index(conn, _params) do
+  render(conn, "index.html")
+end
+```
+
+To:
+```elixir
+def index(conn, _params) do
+  oauth_github_url = ElixirAuthGithub.login_url_with_scope(["user:email"])
+  render(conn, "index.html", [oauth_github_url: oauth_github_url])
+end
+```
+
+### 5.1 Update the `page/index.html.eex` Template
+
+Open the `/lib/app_web/templates/page/index.html.eex` file
+and type (_or paste_) the following code:
+
+```html
+<section class="phx-hero">
+  <h1>Welcome to Awesome App!</h1>
+  <p>To get started, login to your GitHub Account: <p>
+  <a href="<%= @oauth_github_url %>">
+    <img src="https://i.imgur.com/qwoHBIZ.png" alt="Sign in with GitHub" />
+  </a>
+</section>
+```
+
+> **Note**: we are using an `<img>` button for code brevity. <br />
+We suggest you use the `SVG+CSS` approach described
```

Comment:
Thorough! 💛

Reply:
If I don't set the standard I can't expect anyone _else_ to magically go the extra mile. 😉